core/src/error.rs

```diff
@@ -4,155 +4,181 @@
 
 //! The errors that can be emitted in this crate.
 
+use core::fmt::Display;
+
 use alloc::boxed::Box;
 use alloc::string::String;
-use derive_more::{Display, From};
 use subxt_metadata::StorageHasher;
 
 /// The error emitted when something goes wrong.
-#[derive(Debug, Display, From)]
+#[derive(Debug)]
 pub enum Error {
     /// Codec error.
-    #[display(fmt = "Scale codec error: {_0}")]
     Codec(codec::Error),
     /// Metadata error.
-    #[display(fmt = "Metadata Error: {_0}")]
     Metadata(MetadataError),
     /// Storage address error.
-    #[display(fmt = "Storage Error: {_0}")]
     StorageAddress(StorageAddressError),
     /// Error decoding to a [`crate::dynamic::Value`].
-    #[display(fmt = "Error decoding into dynamic value: {_0}")]
     Decode(scale_decode::Error),
     /// Error encoding from a [`crate::dynamic::Value`].
-    #[display(fmt = "Error encoding from dynamic value: {_0}")]
     Encode(scale_encode::Error),
     /// Error constructing the appropriate extrinsic params.
-    #[display(fmt = "Extrinsic params error: {_0}")]
     ExtrinsicParams(ExtrinsicParamsError),
     /// Block body error.
-    #[display(fmt = "Error working with block body: {_0}")]
     Block(BlockError),
 }
 
+impl core::fmt::Display for Error {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Error::Codec(e) => write!(f, "Scale codec error: {e}"),
+            Error::Metadata(e) => write!(f, "Metadata Error: {e}"),
+            Error::StorageAddress(e) => write!(f, "Storage Error: {e}"),
+            Error::Decode(e) => write!(f, "Error decoding into dynamic value: {e}"),
+            Error::Encode(e) => write!(f, "Error encoding from dynamic value: {e}"),
+            Error::ExtrinsicParams(e) => write!(f, "Extrinsic params error: {e}"),
+            Error::Block(e) => write!(f, "Error working with block_body: {}", e),
+        }
+    }
+}
+
 #[cfg(feature = "std")]
 impl std::error::Error for Error {}
 
-impl From<scale_decode::visitor::DecodeError> for Error {
-    fn from(value: scale_decode::visitor::DecodeError) -> Self {
-        Error::Decode(value.into())
-    }
-}
+convert_error!(ExtrinsicParamsError as Error::ExtrinsicParams);
```

Comment:
What do you think about:

```rust
impl_from!(ExtrinsicParamsError => Error::ExtrinsicParams);
```

I find "as" slightly misleading because in my head `as == typecast` such as "usize::MAX as u64".... 

Ideally we could find a similar crate as derive_more or some other crate for error handling where it supports opting of from std::error::Error which is the reason why we are not using thiserror in subxt but this should work as work around for now.

Reply:
changed `convert_from` to `impl_from` with the `as` -> `=>` swap